pkg/apis/batch/types.go

```diff
@@ -475,15 +474,13 @@ type JobStatus struct {
 
 	// The number of pods which are terminating (in phase Pending or Running
 	// and have a deletionTimestamp).
-	// The value is zero (or null) for finished jobs.
```

Comment:
Don't we eventually converge to 0?

Reply:
Unfortunately no, from the example in https://github.com/kubernetes/kubernetes/issues/123775#issuecomment-1982959351, because the job has Failed condition it is no longer reconciled by Job controller, so this is the final status.

Reply:
Ok, so we can't enforce this on the API level, because the pods could be still terminating at the time of updating the final conditions. But wouldn't it make sense to reconcile the job regardless, so we can converge to the zeroed out status? And stop reconciling once there are no pods?

Reply:
I see now that issue and this option is being discussed in https://github.com/kubernetes/kubernetes/issues/123775

Reply:
Correct, we may revisit the comments and the  validation rules when we have consensus. But given it was like that for a long time and the new features is still alpha I think it makes sense to reflect the status quo for 1.30 and make the adjustments in 1.31, so that we don't rush with possibly breaking or risky changes.